### PR TITLE
Adding deploy vcl task

### PR DESCRIPTION
**Pull request by aintgoin2goa**
So apps with their own fastly config can deploy it themselves

### CODE REVIEW

**Comment by matthew-andrews** [comment]
ohhh yes i like this.

**Comment by matthew-andrews** [comment]
i'm surprised it's not complained at you for not updating the readme >_>

**Comment by aintgoin2goa** [comment]
It does locally.  Nice touch :)  Will fix now

**Comment by matthew-andrews** [comment]
yeah md5 doesn't seem availalble in Travis…… might ask them about that……

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "debug": "^2.1.1",
     "denodeify": "^1.2.0",
     "es6-promise": "^2.0.1",
+    "fastly": "^1.2.1",
```
Comment (matthew-andrews) [inline package.json]: think i might instigate a one in one out policy for dependencies…… don't want an Origami Build Tools 2 minute installation……

File: package.json [hunk 2]
```diff
@@ -18,7 +19,8 @@
     "heroku-client": "^1.9.1",
     "isomorphic-fetch": "^2.0.0",
     "origami-build-tools": "^2.13.0",
-    "semver": "^4.3.1"
+    "semver": "^4.3.1",
+    "fastly": "git+https://github.com/Financial-Times/fastly.git#v2.5.0"
```
Comment (matthew-andrews) [inline package.json]: seems duplicated……

also

`Financial-Times/fastly#v2.5.0` cleaner :wink:

Comment (aintgoin2goa) [inline package.json]: Thanks will fix.  Also,  according to npm-check I can get rid of debug and heroku-client.  does that sound right?  I know you hate debug and I guess heroku-client is a dep of haikro anyway

Comment (matthew-andrews) [inline package.json]: i don't hate debug, i just don't use it

Comment (matthew-andrews) [inline package.json]: heroku-client isn't a dependency of haikro, haikro just talks to the REST api directly but that sounds good!

**Comment by aintgoin2goa** [comment]
@matthew-andrews Can I merge this now, please?

**Comment by commuterjoy** [comment]
This is nice, but my only advice is to think about idiot proofing this. It's really easy to set an env var, move directory, do a deploy in some other project and overwrite the VCL. Wondering if this might be an issue.

Perhaps specify the service ID as a parameter and leave the API key as an env var.

File: bin/next-build-tools.js [hunk 3]
```diff
@@ -136,6 +137,19 @@ program
 		});
 
 	program
+		.command('deploy-vcl [folder]')
+		.description('Deploys VCL in [folder] to the fastly service given in the FASTLY_SERVICE_ID env var.  Also requires FASTLY_KEY env var')
+		.option('-m, --main <main', 'Set the name of the main vcl file (the entry point).  Defaults to "main.vcl"')
```
Comment (matthew-andrews) [inline bin/next-build-tools.js]: why not just specify the path to the VCL file directly?  and then default to wherever you think is best to put the main VCL file?

**Comment by aintgoin2goa** [comment]
Good idea - have implemented it as an option to match the heroku cli

File: bin/next-build-tools.js [hunk 4]
```diff
@@ -136,6 +137,20 @@ program
 		});
 
 	program
+		.command('deploy-vcl [folder]')
```
Comment (matthew-andrews) [inline bin/next-build-tools.js]: can we make this `deploy-vcl [file]` would be a bit simpler since this only support a single file anyway

Comment (aintgoin2goa) [inline bin/next-build-tools.js]: No - it supports a whole folder.  The code is taken from fastly-deploy which uploads a bunch of .vcl

Comment (matthew-andrews) [inline bin/next-build-tools.js]: ah ok

**Comment by aintgoin2goa** [comment]
Can I merge now please?

**Comment by matthew-andrews** [comment]
go for it

**Pull request merged**